src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -38961,10 +39066,751 @@ static void F(object? x, object? y)
+                //             t.Item10.ToString(); // 7
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t.Item10").WithLocation(19, 13),
+                // (24,13): warning CS8602: Possible dereference of a null reference.
+                //             t.Rest.Item2.ToString(); // 8
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t.Rest.Item2").WithLocation(24, 13),
+                // (25,13): warning CS8602: Possible dereference of a null reference.
+                //             t.Rest.Item3.ToString(); // 9
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t.Rest.Item3").WithLocation(25, 13));
+            comp.VerifyTypes();
+        }
+
+        [Fact]
+        [WorkItem(29970, "https://github.com/dotnet/roslyn/issues/29970")]
+        public void Tuple_Assignment_10()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F(object x, string y)
+    {
+        (object, string?) t = new ValueTuple<object?, string>(null, """") { Item1 = x }; // 1
+        (object?, string) u = new ValueTuple<object?, string>() { Item2 = y };
+        t.Item1.ToString();
+        t.Item2.ToString();
+        u.Item1.ToString(); // 2
+        u.Item1.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (6,31): warning CS8619: Nullability of reference types in value of type '(object?, string)' doesn't match target type '(object, string?)'.
+                //         (object, string?) t = new ValueTuple<object?, string>(null, "") { Item1 = x }; // 1
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, @"new ValueTuple<object?, string>(null, """") { Item1 = x }").WithArguments("(object?, string)", "(object, string?)").WithLocation(6, 31),
+                // (10,9): warning CS8602: Possible dereference of a null reference.
+                //         u.Item1.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "u.Item1").WithLocation(10, 9));
+            comp.VerifyTypes();
+        }
+
+        [Fact]
+        [WorkItem(29970, "https://github.com/dotnet/roslyn/issues/29970")]
+        public void Tuple_Assignment_11()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F(bool b, object x, object? y)
+    {
+        var t = new ValueTuple<object?, object, object?, object, object?, ValueTuple<object, object?>, object, ValueTuple<object, object?, object>>(
+            x,
+            x,
+            x,
+            y, // 1
+            y,
+            new ValueTuple<object, object?>(y, x), // 2
+            x,
+            new ValueTuple<object, object?, object>(x, y, y)) // 3
+            /*T:(object?, object, object?, object, object?, (object, object?), object, (object?, object, object?))*/;
```

Comment:
>(object?, object, object?) [](start = 87, length = 26)

Could you also try this with the Rest being a 3-tuple `(x, y, y)`?
I think the `T:` should then be a 10-tuple. I'm not sure if it will affect anything else.